lib/charms/layer/jenkins/api.py

```diff
@@ -110,21 +110,18 @@ def delete_node(self, host):
     def reload(self):
         """Reload configuration from disk."""
         hookenv.log("Reloading configuration from disk")
-        client = self._make_client()
-        request = requests.Request("POST", urljoin(self.url, "reload"))
-        try:
-            client.jenkins_open(request)
-        except requests.exceptions.HTTPError as error:
-            # We expect a 'Service Unavailable' error code and to be at the
-            # home page.
-            if error.response.status_code != 503:
-                hookenv.log("Unexpected HTTP response code '%d'" % error.response.status_code)
-                raise
-            if error.response.url != self.url:
-                hookenv.log("Unexpected HTTP response url '%s'" % error.response.url)
-                raise
-        else:
-            raise RuntimeError("Couldn't reload configuration")
+        action = "reload"
+        fail_message = "Couldn't reload configuration"
+        self._execute_action(action, fail_message)
+
+    def restart(self):
+        """Execute a safe restart which waits for all jobs to complete and
+        then restarts"""
```

Comment:
PEP-8 wants this docstring to fit on a single line, """Execute a safe restart, wait for jobs, and restart""" perhaps.